### DISCUSSION TITLE
[Bug] useStore.persist is undefined in React Native

### DISCUSSION THREAD

**Discussion by miikebar** [pmndrs/zustand#1795] [General] [answered]
## Summary

Following the [persist documentation](https://github.com/pmndrs/zustand/blob/main/docs/integrations/persisting-store-data.md) I've been trying to access `useStore.persist.setOptions()` but the `persist` object seems to be undefined in React Native.

The reproduction contains an example `configureBearStore` function which changes the storage name and logs the options to the console. When you run it on the web it works fine, but once you run it on a device in the Expo Snack, a TypeError will be thrown. 

## Error
```
TypeError: Cannot read property 'setOptions' of undefined, js engine: hermes

undefined is not an object (evaluating 'useBearStore.persist.setOptions')
```

![image](https://user-images.githubusercontent.com/9134970/236537963-c718b68a-fec4-4da2-ae82-826668750406.png)

## Link to reproduction

- Web: https://codesandbox.io/p/sandbox/awesome-wescoff-wq9dn5
- React Native: https://snack.expo.dev/@michal.bar/zustand-persist-undefined

## Environment

- Expo: 48
- React Native: 0.71.7
- Zustand: 4.3.8

**Comment by miikebar** [comment, marked as answer]
It looks like this is caused by the fact that if you don't provide the `storage` option when creating the store, then `localStorage` is used by default, which is `undefined` on React Native.

Here's a repro setting a dummy storage implementation: https://snack.expo.dev/@michal.bar/zustand-persist-undefined-custom-storage. Once a storage option has been set the `useBearStore.persist` is no longer `undefined`.

Should the whole `persist` object be undefined because of this? Maybe it's something worth mentioning in the docs? In my case I didn't set the `storage` directly as I needed to set it dynamically based on some conditions using `setOptions`.

**Reply by dai-shi** [reply]
Yeah, docs might not be very friendly to RN users.
Please feel free to improve docs.

> Should the whole `persist` object be undefined because of this?

It seems so:
https://github.com/pmndrs/zustand/blob/639aa00699aaa0e987d605b7a58541b3b186f160/src/middleware/persist.ts#L391-L402

Do you see that warning message?

**Reply by miikebar** [reply]
> Do you see that warning message?

The warning is not logged, I've tried adding some debug logs directly to the installed node_module and it seems that the TypeError is thrown before the function passed to the `config` has a chance to be executed.

```js
if (!storage) {
  console.log('=> No storage defined')
  return config(function () {
    console.log('=> Log warning')
    console.warn("[zustand persist middleware] Unable to update item '" + options.name + "', the given storage is currently unavailable.");
    set.apply(void 0, arguments);
  }, get, api);
}
```

```bash
LOG  => No storage defined
ERROR  TypeError: Cannot read property 'setOptions' of undefined, js engine: hermes
ERROR  Invariant Violation: "main" has not been registered. This can happen if:
* Metro (the local dev server) is run from the wrong folder. Check if Metro is running, stop it and restart it in the current project.
* A module failed to load due to an error and `AppRegistry.registerComponent` wasn't called., js engine: hermes
```

Same thing happens on the web: https://codesandbox.io/p/sandbox/zustand-persist-undefined-forked-mbmj1h?file=%2Fsrc%2FApp.tsx

**Reply by dai-shi** [reply]
Thanks for the investigation.
I think having the warning in the function is intentional, because otherwise, people using SSR use the warning all the time. What do we do? Maybe attach a dummy `persist` to `api`?

**Reply by dbritto-dev** [reply]
On RN, you should use `storage: createJSONStorage(() => AsyncStorage)` but you need to keep in mind these  points about async storages on Zustand
- https://docs.pmnd.rs/zustand/integrations/persisting-store-data#hydration-and-asynchronous-storages
- https://docs.pmnd.rs/zustand/integrations/persisting-store-data#how-can-i-check-if-my-store-has-been-hydrated

**Reply by miikebar** [reply]
@dai-shi
>I think having the warning in the function is intentional, because otherwise, people using SSR use the warning all the time. What do we do? Maybe attach a dummy persist to api?

As the `useStore.persist` is also `undefined` in the SSR environment this raises the same problem ([repro](https://codesandbox.io/p/sandbox/white-sun-sd2m2g?file=%2Fpages%2Findex.tsx)). Is the `storage` required for anything else than `rehydrate` to work? If not, then maybe the `persist` could contain the full implementation but `rehydrate` would throw an error if `storage` is undefined.

---

@dbritto-dev 

Sure, but in my case I'm creating a cross-platform hook that will be used on both web and native, which lead me to not setting the `storage` in the store definition, as I wanted to set it later using `useStore.persist.setOptions({ storage })`. Depending on the environment the storage would be a noop for web (doesn't actually store anything) or `SecureStore` for native.

I've currently fixed the problem by setting the noop storage as a default in the store definition, disabling the automatic hydration, and then setting the desired storage and rehydrating manually later on.

---

I think the question is whether we should treat it as an edge case and just mention it in the documentation, or is it something that should be addressed? This occurs both in RN and SSR if no `storage` is passed.

**Reply by dai-shi** [reply]
> If not, then maybe the persist could contain the full implementation but rehydrate would throw an error if storage is undefined.

Yeah probably, and `set` (and `setState`) should show a warning.
It will be fairly a big change and I'm not sure if it will cause troubles with existing use cases. We should keep watching people reporting issues if we make such changes.
Would you be willing to work on it in a longer term?

**Reply by dbritto-dev** [reply]
@MrPumpking if you want and isomorphic solution you need to write it like this:

```ts
storage: isomorphicStorage
```

```ts
// isomorphic-storage.ts
export const isomorphicStorage = createJSONStorage(() => AsyncStorage)
```

```ts
// isomorphic-storage.web.ts
export const isomorphicStorage = createJSONStorage(() => localStorage)
```

**Reply by miikebar** [reply]
@dai-shi 
> We should keep watching people reporting issues if we make such changes. Would you be willing to work on it in a longer term?

I think that for now updating the documentation to inform users of this issue should suffice, as it can be resolved by simply passing a noop storage. If more bug reports appear then I could try implementing the necessary modifications to solve the problem.

---

@dbritto-dev 

That's also a nice solution, but I'm working with a monorepo and I wanted to avoid including `AsyncStorage` in the package dependencies, so that it's not unnecessarily installed when using the package with web apps.

My current solution is as follows:

`/packages/company-react/usePersistence.ts`

```ts
export const usePersistence = create(
  combine(
    {
      local: null as StateStorage | null,
      secure: null as StateStorage | null,
      noop: {
        getItem: () => null,
        setItem: () => {},
        removeItem: () => {},
      } satisfies StateStorage,
    },
    () => ({})
  )
);

export const configurePersistence = ({
  local,
  secure,
}: ConfigurePersistenceParams) => {
  if (local) {
    usePersistence.setState({ local });
  }
  if (secure) {
    usePersistence.setState({ secure });
  }
};
```

`/packages/company-react/useAuthService.ts`

```ts
export const useAuthService = create(
  persist(
    combine(
      {
        isInitialised: false as boolean,
        accessToken: null as string | null,
        refreshToken: null as string | null,
      },
      (set, get) => ({
        // actions
      })
    ),
    {
      name: 'auth-service',
      version: 0,
      skipHydration: true,
      // We have to pass a default NOOP implementation
      // @see https://github.com/pmndrs/zustand/discussions/1795
      storage: createJSONStorage(() => usePersistence.getState().noop),
      onRehydrateStorage: () => (state) => {
        state?.setInitialised(true);
      },
    }
  )
);

export const configureAuthService = ({
  persistence,
}: ConfigureAuthServiceOptions) => {
  if (persistence) {
    useAuthService.persist.setOptions({
      storage: createJSONStorage(() => persistence),
    });
    useAuthService.persist.rehydrate();
  } else {
    useAuthService.setState({ isInitialised: true });
  }
};
```

`/apps/native/App.tsx`

```ts
import * as SecureStore from 'expo-secure-store';

configurePersistence({
  secure: {
    getItem: SecureStore.getItemAsync,
    setItem: SecureStore.setItemAsync,
    removeItem: SecureStore.deleteItemAsync,
  },
});

configureAuthService({
  persistence: usePersistence.getState().secure,
});
```

This way I can use the `useAuthService` on both platforms. Native persists the tokens in SecureStore, and web doesn't save them at all, as they are saved in HttpOnly cookies, and the token is being provided to the store by `useSession` from NextAuth.

**Reply by dbritto-dev** [reply]
@MrPumpking interesting I was thinking in three repos/monorepos `useAuthService/core`, `useAuthService/react` (react) and `useAuthService/native` (react native).

```ts
const defaultPersistConfig = {...}

// useAuthService/core
export const createUseAuthService = (persistConfig) => create(persist(..., { ...defaultPersistConfig, ...persistConfig }))
```

```ts
const noop = () => {}

const storage = {
  getItem: noop,
  setItem: noop,
  removeItem: noop,
}

// useAuthService/react
export const useAuthService = createUseAuthService({ storage })
```

```ts
import * as SecureStore from 'expo-secure-store'

const storage ={
  getItem: SecureStore.getItemAsync,
  setItem: SecureStore.setItemAsync,
  removeItem: SecureStore.removeItemAsync,
}

// useAuthService/native
export const useAuthService= createUseAuthService({ storage })
```

**Reply by miikebar** [reply]
@dbritto-dev the problem with this approach is that my react package has some hooks that rely on the `useAuthService`, which means I can't have different instances for web and native. I.e I have a hook which fetches data only if the user is authenticated.

Having to create different instances for apps would force me to inject the service using the Context API, and that's also something I wanted to avoid.

**Reply by dbritto-dev** [reply]
> @dbritto-dev the problem with this approach is that my react package has some hooks that rely on the `useAuthService`, which means I can't have different instances for web and native. I.e I have a hook which fetches data only if the user is authenticated.
> 
> Having to create different instances for apps would force me to inject the service using the Context API, and that's also something I wanted to avoid. 

Hmmm but web and native have different instances. It's not like you can share same instance between web and native. My suggestion is not worry about having an extra dep on your `usePersitence` and/or `useAuthService` because expo give you the option to only bundle what you need for web and native.

**Reply by miikebar** [reply]
> Hmmm but web and native have different instances. It's not like you can share same instance between web and native.

That's what I want to avoid. For example

`/packages/company-react/data/useData.ts`

```ts
import { useAuthService } from '../auth';

const useData = () => {
  const isAuthenticated = useAuthService((state) => state.isAuthenticated);

  return useQuery({
    queryFn: () => { ... },
    enabled: isAuthenticated
  })
}
```

If `useAuthService` was initialised directly in apps and not in the react package then I couldn't really use it in custom hooks without injecting it using Context, and then consuming it in hooks (i.e `useAuthContext`). With my approach the `useAuthService` is initialised once (in `/packages/company-react/auth/useAuthService.ts`), and then it's just configured with a different persistence layer depending on the platform: 

`/apps/web/pages/_app.tsx`

```ts
import type { AppProps } from 'next/app';
import { configureAuthService, usePersistence, useData } from '@company/react';

configureAuthService({
  persistence: usePersistence.getState().noop
});

export default function MyApp({ Component, pageProps }: AppProps) {
  const { data } = useData();

  return <Component {...pageProps} />;
}
```

`apps/native/App.tsx`

```ts
import * as SecureStore from 'expo-secure-store';
import { configureAuthService, usePersistence, useData } from '@company/react';

configurePersistence({
  secure: {
    getItem: SecureStore.getItemAsync,
    setItem: SecureStore.setItemAsync,
    removeItem: SecureStore.deleteItemAsync,
  },
});

configureAuthService({
  persistence: usePersistence.getState().secure,
});

const App: FC = () => {
  const { data } = useData();
}

export default App;
```



> My suggestion is not worry about having an extra dep on your usePersitence and/or useAuthService because expo give you the option to only bundle what you need for web and native.

I mainly wanted to avoid downloading native packages in CI builds. I know they can be cached but preview builds with no cache took longer the more native modules I had (at least that was my experience with Nx monorepos on Vercel, not sure if it would be the case with Turbo). I don't use Expo nor react-native-web to build the web apps - I share the logic but have separate UIs.

**Comment by dai-shi** [comment]
https://github.com/pmndrs/zustand/discussions/1795#discussioncomment-5824284
Improving docs for RN users sounds like a great idea. Please feel free to open a PR.

**Comment by Hamzachikhaoui123** [comment]
Hi everyone,
I'm facing an issue with my React Native app using Redux. When I try to run the app, I get the following error: 
TypeError: Cannot read property 'store' of undefined
![Screenshot_2025-10-14-11-49-02-68_fdefcd99393dad248a8c8f6719c96011](https://github.com/user-attachments/assets/167cd32f-b0ac-47fd-b4bd-65b7daa295b5)

/* eslint-disable linebreak-style */
import React, { useEffect } from "react";
import { NativeModules, Platform, StatusBar } from "react-native";
import {
	NavigationContainer,
	DefaultTheme,
	useTheme
} from "@react-navigation/native";

// Redux
import { PersistGate } from "redux-persist/es/integration/react";
import { Provider  } from "react-redux";
import { store, persistor } from "./redux/store";
import { AuthCheck } from "./routes";


const MyTheme = {
	dark: false,
	...DefaultTheme,
	colors: {
		...DefaultTheme.colors,
		primary: "#FFFFFF",
		background: "#ededed",
		button: "#007481",
		buttonText: "#FFFFFF",
		link: "blue",
		text: "#000000",
		inputBorder: "#adacac",
		red: "red",
		green:"#48b9b2",
		transparent_gray: "#000000AA",
		ed_gray: "#d9d9d9",
		opal: "#9ec1c9",
		opal_dark: "#008d93",
		opal_light: "#badfdc",
		pink : "#b1005a",
		colorLogo:"#820040"

	}
};

const App = () => {
	// const { colors } = useTheme();
	// useEffect(() => {
	// 	if (Platform.OS === "ios") {
	// 		SplashScreen.hide();
	// 	} else if (Platform.OS === "android") {
	// 		NativeModules.SplashScreen.hide();
	// 	}
	// }, []);



  
	

	return (
	<Provider store={store}>
			<PersistGate loading={null} persistor={persistor}>
				<NavigationContainer theme={MyTheme} testID="app-container">
					<StatusBar
						testID="status-bar"
						animated={true}
						backgroundColor={colors.background}
						barStyle={"dark-content"}
						hidden={false}
					/>
					<AuthCheck testID="auth-check" />
				</NavigationContainer>
			</PersistGate>
		</Provider>
	

)}
export default App;
import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import createSagaMiddleware from "redux-saga";

import { persistStore, persistReducer } from 'redux-persist';

import AsyncStorage from '@react-native-async-storage/async-storage';

import { encryptTransform } from 'redux-persist-transform-encrypt';

import authReducer from './auth/reducer';
import patientsReducer from './patients/reducer';
import rendezVousReducer from './rendez-vous-doctor/reducer';
import doctorsServicesSaga from './doctors/saga'
import authServicesSaga from './auth/saga';
import patientsServicesSaga from './patients/saga';
import rendezVousServicesSaga from './rendez-vous-doctor/saga';
import doctorReducer from './doctors/reducer';
import specialitieReducer from './specialities/reducer';
import specialiteServicesSage from './specialities/saga';
import motifsReducer from './motifs/reducer';
import registerReducer from './register/reducer';
import registerSaga from './register/saga'
import motifsSaga from './motifs/saga';
import rendezVousPatientReducer from './rendez-vous-patients/reducer'
import rendezVousPatientServicesSaga from './rendez-vous-patients/saga'
import secretarySaga from './secretaire/saga'
import secretaryReducer from './secretaire/reducer'
import rendezVousSecretaireReducer from './rendez-vous-secretaire/reducer'
import rendezVousSecretaireServicesSaga from './rendez-vous-secretaire/saga'
import accompanimentsSaga from './patient-accompaniments/saga'
import fileDoctorSaga from './files/saga'
import fileReducer from './files/reducer'
import filePatientReducer from './filesPatient/reducer'
import filePatientSaga from './filesPatient/saga'
import consultationReducer from './consultation/reducer';
import patientsAccompanimentsReducer from './patient-accompaniments/reducer'
import ConsutationsSaga from './consultation/saga';
import StaticRdvReducer from './static-rdv/reduce';
import StaticRdvSaga from './static-rdv/saga';
import StaticPatientReducer from './static-patient/reduce';
import StaticPatientSaga from './static-patient/saga';
import ProjectionRdvReducer from './projection-rdv/reduce';
import ProjectionRdvSaga from './projection-rdv/saga';
import ProjectionPatientReducer from './projection-patient/reduce';
import ProjectionPatientSaga from './projection-patient/saga';
import CheckPhoneReducer from './check-phone/reduce';
import CheckPhoneSaga from './check-phone/saga';
import fileDoctorPatientReducer from './DocumentsDoctor-Patient/reducer';
import FileDoctorPatientSaga from './DocumentsDoctor-Patient/saga';
import paiementReducer from './paiement/reducer';
import paiementSaga from './paiement/saga';
import paiementSecSaga from './paiement-sec/saga';
import paiementSecReducer from './paiement-sec/reducer';
import paiementPatientReducer from './paiement-patient/reducer';
import paiementPatientSaga from './paiement-patient/saga';
import hisotrieconsultationReducer from './hisotrieConsultation/reducer';
import hisotrieConsutationsSaga from './hisotrieConsultation/saga';
import balanceReducer from './balance-doctor/reducer';
import balanceSaga from './balance-doctor/saga';
import parametreSanteReducer from './parametre-sante/reducer';
import parametreSanteSaga from './parametre-sante/saga';
import biologieReducer from './biologie/reducer';
import biologieSaga from './biologie/saga';
import radiologieSaga from './radiologie/saga';
import radioReducer from './radiologie/reducer';
import actesReducer from './actes/reducer';
import actesSaga from './actes/saga';
import sendEmailReducer from './send-email/reducer';
import sendEmailSaga from './send-email/saga';
import confrimReducer from './confrim-patient/reducer';
import confrimSaga from './confrim-patient/saga';
import languesReducer from './langue/reducer';
import languesSaga from './langue/saga';
import prescriptionsReducer from './prescriptions/reducer';
import prescriptionsServicesSaga from './prescriptions/saga';
import CallVideoSdkDoctorReducer from './video-sdk-doctor/reduce';
import callVideoSdkDoctorSaga from './video-sdk-doctor/saga';
import videoSdkPatientReducer from './video-sdk-patient/reduce';
import videoSdkPatientSaga from './video-sdk-patient/saga';
let composeEnhancers = compose;

if (__DEV__) {
  composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
}

const encryptor = encryptTransform({
  secretKey: 'Siladoc',
  onError: function (error) {
  },
})

const rootReducer = combineReducers({
  authReducer,
  patientsReducer,
  rendezVousReducer,
  doctorReducer,
  specialitieReducer,
  registerReducer,
  motifsReducer,
  rendezVousPatientReducer,
  secretaryReducer,
  rendezVousSecretaireReducer,
  patientsAccompanimentsReducer,
  fileReducer,
  filePatientReducer,
  consultationReducer,
  StaticRdvReducer,
  StaticPatientReducer,
  ProjectionRdvReducer,
  ProjectionPatientReducer,
  CheckPhoneReducer,
  fileDoctorPatientReducer,
  paiementReducer,
  paiementSecReducer,
  paiementPatientReducer,
  hisotrieconsultationReducer,
  balanceReducer,
  parametreSanteReducer,
  biologieReducer,
  radioReducer,
  actesReducer,
  sendEmailReducer,
  confrimReducer,
  languesReducer,
  prescriptionsReducer,
  CallVideoSdkDoctorReducer,
  videoSdkPatientReducer
})

// Middleware: Redux Persist Config
const persistConfig = {
  // Root?
  key: 'root',
  // Storage Method (React Native)
  storage: AsyncStorage,

  transforms: [encryptor],
  // Whitelist (Save Specific Reducers)
  whitelist: ['authReducer'],
  blacklist: []
};

const sagaMiddleware = createSagaMiddleware();

// Middleware: Redux Persist Persisted Reducer
const persistedReducer = persistReducer(persistConfig, rootReducer)

const store = createStore(persistedReducer, composeEnhancers(applyMiddleware(sagaMiddleware)));

sagaMiddleware.run(authServicesSaga);
sagaMiddleware.run(patientsServicesSaga);
sagaMiddleware.run(rendezVousServicesSaga);
sagaMiddleware.run(doctorsServicesSaga);
sagaMiddleware.run(specialiteServicesSage);
sagaMiddleware.run(motifsSaga);
sagaMiddleware.run(rendezVousPatientServicesSaga);
sagaMiddleware.run(registerSaga)
sagaMiddleware.run(secretarySaga)
sagaMiddleware.run(rendezVousSecretaireServicesSaga)
sagaMiddleware.run(accompanimentsSaga)
sagaMiddleware.run(fileDoctorSaga)
sagaMiddleware.run(filePatientSaga)
sagaMiddleware.run(ConsutationsSaga)
sagaMiddleware.run(StaticRdvSaga)
sagaMiddleware.run(StaticPatientSaga)
sagaMiddleware.run(ProjectionRdvSaga)
sagaMiddleware.run(ProjectionPatientSaga)
sagaMiddleware.run(CheckPhoneSaga)
sagaMiddleware.run(FileDoctorPatientSaga)
sagaMiddleware.run(paiementSaga)
sagaMiddleware.run(paiementSecSaga)
sagaMiddleware.run(paiementPatientSaga)
sagaMiddleware.run(balanceSaga)
sagaMiddleware.run(parametreSanteSaga)
sagaMiddleware.run(hisotrieConsutationsSaga)
sagaMiddleware.run(biologieSaga)
sagaMiddleware.run(radiologieSaga)
sagaMiddleware.run(actesSaga)
sagaMiddleware.run(sendEmailSaga)
sagaMiddleware.run(confrimSaga)
sagaMiddleware.run(languesSaga)
sagaMiddleware.run(prescriptionsServicesSaga)
sagaMiddleware.run(callVideoSdkDoctorSaga)
sagaMiddleware.run(videoSdkPatientSaga)
// Middleware: Redux Persist Persister
let persistor = persistStore(store);

export {
  store,
  persistor
};

**Reply by Hamzachikhaoui123** [reply]
"dependencies": {
    "@react-native-async-storage/async-storage": "^2.2.0",
    "@react-native-community/checkbox": "^0.5.20",
    "@react-native-picker/picker": "^2.11.2",
    "@react-navigation/drawer": "^7.5.9",
    "@react-navigation/elements": "^2.6.5",
    "@react-navigation/native": "^7.1.18",
    "@react-navigation/native-stack": "^7.3.28",
    "@react-navigation/stack": "^7.4.9",
    "axios": "^1.12.2",
    "date-fns": "^4.1.0",
    "deprecated-react-native-prop-types": "^5.0.0",
    "lodash": "^4.17.21",
    "lottie-react-native": "^7.3.4",
    "moment": "^2.30.1",
    "npm": "^11.6.2",
    "react": "19.1.1",
    "react-native": "0.82.0",
    "react-native-blob-util": "^0.22.2",
    "react-native-calendar-strip": "^2.2.6",
    "react-native-calendars": "^1.1313.0",
    "react-native-chart-kit": "^6.12.0",
    "react-native-confirmation-code-field": "^8.0.1",
    "react-native-dotenv": "^3.4.11",
    "react-native-gesture-handler": "^2.28.0",
    "react-native-hyperlink": "^0.0.22",
    "react-native-image-picker": "^8.2.1",
    "react-native-multi-selectbox": "^1.5.0",
    "react-native-pdf": "^7.0.1",
    "react-native-permissions": "^5.4.2",
    "react-native-safe-area-context": "^5.6.1",
    "react-native-screens": "^4.16.0",
    "react-native-timeline-flatlist": "^0.8.0",
    "react-native-vector-icons": "^10.3.0",
    "react-native-video": "^6.17.0",
    "react-native-vision-camera": "^4.7.2",
    "react-native-webview": "^13.16.0",
    "react-redux": "^8.0.5",
    "redux": "^4.2.0",
    "redux-persist": "^6.0.0",
    "redux-persist-transform-encrypt": "^3.0.1",
    "redux-saga": "^1.2.2"
  },
  "devDependencies": {
    "@babel/core": "^7.25.2",
    "@babel/preset-env": "^7.25.3",
    "@babel/runtime": "^7.25.0",
    "@react-native-community/cli": "20.0.0",
    "@react-native-community/cli-platform-android": "20.0.0",
    "@react-native-community/cli-platform-ios": "20.0.0",
    "@react-native/babel-preset": "0.82.0",
    "@react-native/eslint-config": "0.82.0",
    "@react-native/metro-config": "0.82.0",
    "@types/jest": "^29.5.13",
    "@types/react": "^19.1.1",
    "@types/react-test-renderer": "^19.1.0",
    "eslint": "^8.19.0",
    "jest": "^29.6.3",
    "metro-react-native-babel-preset": "^0.77.0",
    "prettier": "2.8.8",
    "react-test-renderer": "19.1.1",
    "redux-logger": "^3.0.6",
    "typescript": "^5.8.3"
  },
  "engines": {
    "node": ">=20"
  }
}
